.circleci/config.yml

```diff
@@ -1,37 +1,118 @@
 version: 2.1
 
 orbs:
-  cli: circleci/circleci-cli@0.1.4
+  php: circleci/php@<<pipeline.parameters.dev-orb-version>>
+  orb-tools: circleci/orb-tools@9.1
 
-workflows:
-  main:
-    jobs:
-      - test:
-          filters:
-            tags:
-              # Simplified SemVer regex
-              only: /^\d+\.\d+\.\d+$/
-      - publish:
-          filters:
-            branches:
-              ignore: /.*/
-            tags:
-              # Simplified SemVer regex
-              only: /^\d+\.\d+\.\d+$/
-          context: orb-publishing
+parameters:
+  run-integration-tests:
+    type: boolean
+    default: false
+  dev-orb-version:
+    type: string
+    default: "dev:alpha"
 
 jobs:
-  test:
-    executor: cli/default
+  integration_install-php:
+    docker:
+      - image: cimg/base:edge
     steps:
-      - checkout
-      - run:
-          name: "Validate Orb config"
-          command: circleci orb validate orb.yml
-  publish:
-    executor: cli/default
+      - php/install-php
+
+  integration_install-composer:
+    executor: php/default
+    steps:
+      - php/install-composer
+
+  integration_install-packages:
+    executor: php/default
     steps:
       - checkout
-      - run:
-          name: "Publish orb via the CircleCI CLI"
-          command: circleci orb publish orb.yml circleci/php@${CIRCLE_TAG}
+      - php/install-packages:
+          app-dir: "~/project/sample"
+          cache-key: "composer.json"
+
+# yaml anchor filters
+prod-deploy_requires: &prod-deploy_requires
+  [
+    integration_install-php
+    integration_install-composer
+    integration_install-packages
+  ]
+
+integration-dev_filters: &integration-dev_filters
+  branches:
+    ignore: /.*/
+  tags:
+    only: /integration-.*/
+
+integration-master_filters: &integration-master_filters
+  branches:
+    ignore: /.*/
+  tags:
+    only: /master-.*/
+
+workflows:
+  # This `lint-pack_validate_publish-dev` workflow will run on any commit.
+  lint_pack-validate_publish-dev:
+    unless: << pipeline.parameters.run-integration-tests >>
+    jobs:
+      - orb-tools/lint
+      # pack your orb YAML files to a single orb.yml
+      # validate the orb.yml file to ensure it is well-formed
+      - orb-tools/pack:
+          requires:
+            - orb-tools/lint
+
+      # release dev version of orb, for testing & possible publishing.
+      # orb will be published as dev:alpha and dev:${CIRCLE_SHA1:0:7}.
+      # requires a CircleCI API token to be stored as CIRCLE_TOKEN (default)
+      # https://circleci.com/docs/2.0/managing-api-tokens
+      # store CIRCLE_TOKEN as a project env var or Contexts resource
+      # if using Contexts, add your context below
+      - orb-tools/publish-dev:
```

Comment:
Any job that publishes will need access to a context with credentials.
Example from node: https://github.com/CircleCI-Public/node-orb/blob/master/.circleci/config.yml#L153